utils/src/main/java/com/cloud/utils/nio/NioConnection.java

```diff
@@ -202,57 +237,64 @@ protected void accept(final SelectionKey key) throws IOException {
+                    sslEngine.beginHandshake();
+                    if (!Link.doHandshake(socketChannel, sslEngine, getSslHandshakeTimeout())) {
+                        throw new IOException("SSL handshake timed out with " + socketAddress);
+                    }
+                    if (logger.isTraceEnabled()) {
+                        logger.trace("SSL: Handshake done");
                     }
+                    final Link link = new Link(socketAddress, nioConnection);
+                    link.setSSLEngine(sslEngine);
+                    link.setKey(socketChannel.register(key.selector(), SelectionKey.OP_READ, link));
+                    final Task task = _factory.create(Task.Type.CONNECT, link, null);
+                    registerLink(socketAddress, link);
+                    _executor.submit(task);
+                } catch (final GeneralSecurityException | IOException e) {
+                    if (logger.isTraceEnabled()) {
+                        logger.trace(socket.getRemoteSocketAddress()+ "Connection closed due to failure: " + e.getMessage());
+                    }
+                    closeAutoCloseable(socket, "accepting socket");
+                    closeAutoCloseable(socketChannel, "accepting socketChannel");
+                } finally {
+                    int connections = activeAcceptConnections.decrementAndGet();
+                    if (logger.isTraceEnabled()) {
+                        logger.trace(String.format("Accept task complete for %s - time taken: %d, " +
+                                        "active accept connections: %d",
+                                socketAddress, (System.currentTimeMillis() - startTime), connections));
+                    }
+                    _selector.wakeup();
                 }
             });
-        } catch (final Exception e) {
+        } catch (final RejectedExecutionException e) {
             if (logger.isTraceEnabled()) {
-                logger.trace("Connection closed due to failure: " + e.getMessage());
+                logger.trace("{} Accept Task rejected: {}", socket.getRemoteSocketAddress(), e.getMessage());
             }
-            closeAutoCloseable(socket, "accepting socket");
-            closeAutoCloseable(socketChannel, "accepting socketChannel");
+            closeAutoCloseable(socket, "Rejecting connection - accepting socket");
+            closeAutoCloseable(socketChannel, "Rejecting connection - accepting socketChannel");
         } finally {
             _selector.wakeup();
         }
     }
 
-    protected void terminate(final SelectionKey key) {
+    protected void terminate(final SelectionKey key, String msg) {
         final Link link = (Link)key.attachment();
         closeConnection(key);
         if (link != null) {
+            if (logger.isTraceEnabled()) {
+                logger.warn("Will terminate connection due to: " + msg);
+            }
```

Comment:
Same here, no need for a guard
```suggestion
            logger.warn("Will terminate connection due to: {}", msg);
```